File: app/src/main/java/com/zacharee1/systemuituner/misc/RecreateHandler.java
package com.zacharee1.systemuituner.misc;

import android.app.Activity;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.support.v4.content.LocalBroadcastManager;

import com.zacharee1.systemuituner.activites.SettingsActivity;

public class RecreateHandler
{
    private static BroadcastReceiver mMessageReceiver;

    public static void onCreate(final Activity activity) {
        mMessageReceiver = new BroadcastReceiver() {
            @Override
            public void onReceive(Context context, Intent intent) {
                activity.recreate();
            }
        };

        LocalBroadcastManager.getInstance(activity).registerReceiver(mMessageReceiver,
                new IntentFilter(SettingsActivity.RECREATE_ACTIVITY));
    }

    public static void onDestroy(Activity context) {
        if (mMessageReceiver != null) LocalBroadcastManager.getInstance(context).unregisterReceiver(mMessageReceiver);
    }
}

File: app/src/main/java/com/zacharee1/systemuituner/handlers/RecreateHandler.kt
package com.zacharee1.systemuituner.handlers

import android.app.Activity
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.support.v4.content.LocalBroadcastManager

import com.zacharee1.systemuituner.activites.settings.SettingsActivity

object RecreateHandler {
    private var mMessageReceiver: BroadcastReceiver? = null

    fun onCreate(activity: Activity) {
        mMessageReceiver = object : BroadcastReceiver() {
            override fun onReceive(context: Context, intent: Intent) {
                activity.recreate()
            }
        }

        LocalBroadcastManager.getInstance(activity).registerReceiver(mMessageReceiver!!,
                IntentFilter(SettingsActivity.RECREATE_ACTIVITY))
    }

    fun onDestroy(context: Activity) {
        if (mMessageReceiver != null) LocalBroadcastManager.getInstance(context).unregisterReceiver(mMessageReceiver!!)
    }
}
